libs/output-mapping/src/Writer/Table/TableConfigurationValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\Writer\Table;
+
+use InvalidArgumentException;
+use Keboola\OutputMapping\Exception\InvalidOutputException;
+use Keboola\OutputMapping\Mapping\MappingFromRawConfigurationAndPhysicalDataWithManifest;
+use Keboola\OutputMapping\Writer\Helper\RestrictedColumnsHelper;
+use Keboola\OutputMapping\Writer\Table\Strategy\SqlWorkspaceTableStrategyNew;
+
+class TableConfigurationValidator
+{
+    public function validate(
+        StrategyInterfaceNew $strategy,
+        MappingFromRawConfigurationAndPhysicalDataWithManifest $source,
+        array $config,
+    ): array {
+        if (!$strategy instanceof SqlWorkspaceTableStrategyNew) {
+            try {
+                RestrictedColumnsHelper::validateRestrictedColumnsInConfig(
+                    $config['columns'],
+                    $config['column_metadata'],
+                );
+            } catch (InvalidOutputException $e) {
+                throw new InvalidOutputException(
+                    sprintf(
+                        'Failed to process mapping for table %s: %s',
+                        $source->getSourceName(),
+                        $e->getMessage(),
+                    ),
+                    0,
+                    $e,
+                );
+            }
+        }
+
+        $hasColumns = $config['columns'];
+        if (!$hasColumns && $source->isSliced()) {
+            throw new InvalidOutputException(
+                sprintf('Sliced file "%s" columns specification missing.', $source->getSourceName()),
+            );
+        }
+
+        try {
+            $config['destination'] = new MappingDestination($config['destination']);
```

Comment:
ten MappingDestination se ale jednak bude mazat a jednak je to divný ne, že to vrací assoc pole (dál se imho všude čeká, že to bude assoc pole) a jedna z položek je classa